keps/sig-node/20190717-seccomp-ga.md

```diff
@@ -387,9 +486,226 @@ alpha](#alternatives).
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable, can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/20190731-production-readiness-review-process.md
+
+Production readiness review questionnaire must be completed for features in
+v1.19 or later, but is non-blocking at this time. That is, approval is not
+required in order to be in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+
+-->
+
+### Feature enablement and rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  
+  This feature was implemented prior to the concept of feature gates. Meaning, it is already enabled 
+  in live clusters and cannot be disabled. This KEP is simply "cleaning up" the API to make it GA.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we rollback
+  the enablement)?**
+
+  This feature (seccomp) cannot be disabled, as it is currently in a "pseudo-GA" state. 
+  However, the changes it brings are backwards compatible, and the API supports rollback 
+  of the kubernetes apiserver as described in the [Version Skew Strategy](#version-skew-strategy).
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+  
+  N/A - the feature is already enabled by default since Kubernetes 1.3.
+
+* **Are there any tests for feature enablement/disablement?**
+  The e2e framework does not currently support enabling and disabling feature
+  gates. However, unit tests in each component dealing with managing data created
+  with and without the feature are necessary. At the very least, think about
+  conversion tests if API types are being modified.
+
+  N/A - the feature is already enabled by default since Kubernetes 1.3.
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  Try to be as paranoid as possible - e.g. what if some components will restart
+  in the middle of rollout?
+
+  The [Version Skew Strategy](#version-skew-strategy) section covers this point.
+  Running workloads should have no impact as the Kubelet will support either the 
+  existing annotations or the new fields introduced by this KEP. 
```

Comment:
I'd like clarification on the question at https://github.com/kubernetes/enhancements/pull/1747#discussion_r427406060 for this scenario:

1. A pod with seccomp annotations is created against a 1.18 API server (so no seccomp field is populated) and remains pending
2. The API server is upgraded to 1.19
3. The nodes are upgraded to 1.19
4. The pending pod is scheduled to a 1.19 node
5. Does the 1.19 kubelet honor the seccomp annotation?

Reply:
Yes, the kubelet will continue to honor seccomp annotations until their support is removed later down the line. Until then, the kubelet will accept pods with any of the settings below:

- Seccomp annotations.
- Seccomp annotations and seccomp fields: provided they have the same value.
- Seccomp fields.

Reply:
It wasn't clear to me that the kubelet also would pay attention to both. This doc should describe the order in which the kubelet looks at those to determine the seccomp profile for a container, e.g.:

1. container-specific field
2. container-specific annotation
3. pod-wide field
4. pod-wide annotation

I'm not sure what I would expect the kubelet to do if the field and annotation both exist and disagree. My first instinct would be for the kubelet to use a precedence order and take the first one it found, rather than fail. The only way I can see for the field and annotation to both exist and be different is if the pod's annotation was updated after creation... I would have expected the kubelet to use the field in that case, not fail on the diff.

Reply:
Added a section to cover this points too.